patchwork-fml/src/main/java/net/minecraftforge/forgespi/language/ModFileScanData.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2020, 2019-2020
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.forgespi.language;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.annotation.Annotation;
+import java.lang.annotation.ElementType;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.gson.Gson;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.objectweb.asm.Type;
+
+import net.fabricmc.loader.FabricLoader;
+import net.fabricmc.loader.api.ModContainer;
+import net.fabricmc.loader.api.metadata.CustomValue;
+import net.fabricmc.loader.metadata.LoaderModMetadata;
+import net.fabricmc.loader.metadata.NestedJarEntry;
+
+public class ModFileScanData {
+	private static final Logger LOGGER = LogManager.getLogger();
+
+	private String modid;
+	private boolean initialized = false;
+	private AnnotationStorage annotationStorage;
+	private Set<AnnotationData> annotationData;
+
+	public ModFileScanData(String modid) {
+		this.modid = modid;
+	}
+
+	private void init() {
+		initialized = true;
+
+		ModContainer modContainer = getModContainer(modid);
+		CustomValue customValue = modContainer.getMetadata().getCustomValue("patchwork:annotations");
+
+		if (customValue == null) {
+			LOGGER.error("ModFileScanData is being accessed but cannot find annotation storage");
```

Comment:
Should print the modid here for better debugging
```suggestion
			LOGGER.error("ModFileScanData: Tried to access the scanned annotation data for " + modid + ", but it is missing");
```